src/UCTNode.cpp

```diff
@@ -97,31 +97,24 @@ bool UCTNode::create_children(std::atomic<int> & nodecount,
     m_is_expanding = true;
     lock.unlock();
 
-    auto raw_netlist = Network::get_scored_moves(
+    const auto raw_netlist = Network::get_scored_moves(
         &state, Network::Ensemble::RANDOM_ROTATION);
 
     // DCNN returns winrate as side to move
     auto net_eval = raw_netlist.second;
-    auto to_move = state.board.get_to_move();
+    const auto to_move = state.board.get_to_move();
     // our search functions evaluate from black's point of view
     if (to_move == FastBoard::WHITE) {
         net_eval = 1.0f - net_eval;
     }
     eval = net_eval;
 
-    FastBoard & board = state.board;
     std::vector<Network::scored_node> nodelist;
 
     auto legal_sum = 0.0f;
     for (auto& node : raw_netlist.first) {
         auto vertex = node.second;
-        if (vertex != FastBoard::PASS) {
```

Comment:
I'm not sure where this checks that the vertex isn't occupied, probably a side effect of `!board.is_suicide` but I didn't find an obvious check.

Either way fastfailing if square is occupied (in is_move_legal) is probably better

Reply:
https://github.com/gcp/leela-zero/blob/1894e17d8165b8f88b03f704155400091c07db7b/src/Network.cpp#L459